xclim/indices/_agro.py

```diff
@@ -820,3 +827,708 @@ def effective_growing_degree_days(
+        {s_thresh_dry} for {s_dry_days} consecutive days.
+    s_start_date: str
+        First day of year where season can start ("mm-dd").
+    s_end_date: str
+        Last day of year where season can start ("mm-dd").
+    e_op : str
+        Resampling operator = {"max", "sum", "etp}
+        If "max": based on the occurrence (or not) of an event during the last days of a rain season.
+            The rain season stops when no daily precipitation greater than {e_thresh} have occurred over a period of
+            {e_window} days.
+        If "sum": based on a total amount of precipitation received during the last days of the rain season.
+            The rain season stops when the total amount of precipitation is less than {e_thresh} over a period of
+            {e_window} days.
+        If "etp": calculation is based on the period required for a water column of height {e_thresh] to evaporate,
+            considering that any amount of precipitation received during that period must evaporate as well. If {etp} is
+            not available, the evapotranspiration rate is assumed to be {e_etp_rate}.
+    e_thresh : str
+        Maximum or accumulated precipitation threshold associated with {e_window}.
+        If {e_op} == "max": maximum daily precipitation  during a period of {e_window} days.
+        If {e_op} == "sum": accumulated precipitation over {e_window} days.
+        If {e_op} == "etp": height of water column that must evaporate.
+    e_window: int
+        If {e_op} in ["max", "sum"]: number of days used to verify if the rain season is ending.
+    e_etp_rate: str
+        If {e_op} == "etp": evapotranspiration rate.
+        Otherwise: not used.
+    e_start_date: str
+        First day of year at or after which the season can end ("mm-dd").
+    e_end_date: str
+        Last day of year at or before which the season can end ("mm-dd").
+    """
+
+    def rename_dimensions(
+        da: xarray.DataArray, lat_name: str = "latitude", lon_name: str = "longitude"
+    ) -> xarray.DataArray:
+
+        if ("location" not in da.dims) and (
+            (lat_name not in da.dims) or (lon_name not in da.dims)
+        ):
+            if "dim_0" in list(da.dims):
+                da = da.rename({"dim_0": "time"})
+                da = da.rename({"dim_1": lat_name, "dim_2": lon_name})
+            elif ("lat" in list(da.dims)) or ("lon" in list(da.dims)):
+                da = da.rename({"lat": lat_name, "lon": lon_name})
+            elif ("rlat" in list(da.dims)) or ("rlon" in list(da.dims)):
+                da = da.rename({"rlat": lat_name, "rlon": lon_name})
+            elif (lat_name not in list(da.dims)) and (lon_name not in list(da.dims)):
+                if lat_name == "latitude":
+                    da = da.expand_dims(latitude=1)
+                if lon_name == "longitude":
+                    da = da.expand_dims(longitude=1)
+        return da
```

Comment:
Hum, why is this necessary ? I don't think xclim should be responsible for this.